Sultiple multiple hostnames.

jankeromnes opened the pull request.

notriddle commented:
[Don't use .io](https://getstream.io/blog/stop-using-io-domain-names-for-production-traffic/)

Have you considered using a .tech domain? It's a reasonable length, and it's not being run by a government branch that has no idea what they're doing.

jankeromnes commented:
> /home/travis/build/JanitorTechnology/janitor/join.js
>  255:5  error  Unexpected lexical declaration in case block  no-case-declarations

For real? 😐

jankeromnes commented:
@bnjbvr If you have some time, could you please have a look at these changes?

I basically:
- Removed unused code
- Replaced `"hostname": "single.hostname.com"` with `"hostnames": [ "multiple.com", "hostnames.com" ]` in `db.json`

This is in order to use [janitor.technology](https://janitor.technology) and [jntr.io](http://jntr.io) interchangeably.

notriddle commented:
In the interest of SEO, `<link rel="canonical" href="{{hostnames[0]}}/{{path}}">` seems like something you'd need?

jankeromnes commented:
Great catch @notriddle, thanks! I'll add that to our header templates.

jankeromnes commented:
@bnjbvr or @notriddle, I've rebased this change, and added canonical URLs to every page.

Please have a look whenever it's convenient for you. 😄

bnjbvr commented on app.js:
```diff
@@ -38,13 +38,13 @@ boot.executeInParallel([
+    // Only accept requests addressed to our actual hostnames.
     const requestedHostname = request.headers.host;
-    if (requestedHostname !== hostname) {
+    if (!requestedHostname || hostnames.indexOf(requestedHostname) < 0) {
```
nit: `!hostnames.includes(requestedHostname)`

bnjbvr commented on join.js:
```diff
@@ -14,15 +14,15 @@ const proxyHeuristics = require('./lib/proxy-heuristics');
-if (!hostname || hostname === 'localhost') {
-  throw new Error('Cannot join cluster as [hostname = ' + hostname + ']: ' +
+if (!hostnames || hostnames[0] === 'localhost') {
+  throw new Error('Cannot join cluster as [hostname = ' + hostnames[0] + ']: ' +
```
nit: here and below you could use template strings

bnjbvr replied:
A suggestion: instead of having this file doing this check, what about having a helper somewhere else (any `utils`-like file) that has a `function mainHostname() {}`, and moving the assert in this same file? It would be less custom than all the `hostnames[0]` scattered in all the files, in case the way to store the main hostname ever changes.

bnjbvr commented on lib/certificates.js:
```diff
@@ -90,10 +90,12 @@ exports.isValid = function (parameters, enableDebug) {
-        '(should be one of: "' + altNames.join('", "') + '").');
-      return false;
+    for (const hostname of hostnames) {
+      if (altNames.indexOf(hostname) < 0) {
```
nit: `!altNames.includes(hostname)`
